src/coreclr/vm/jitinterface.cpp

```diff
@@ -1636,6 +1645,126 @@ uint32_t CEEInfo::getThreadLocalFieldInfo (CORINFO_FIELD_HANDLE  field, bool isG
     return typeIndex;
 }
 
+#ifndef _MSC_VER
+
+#ifdef HOST_AMD64
+
+#ifdef TARGET_OSX
+void* getThreadStaticDescriptor(uint8_t* p)
+{
+    _ASSERTE_MSG((p[0] == 0x48 && p[1] == 0x8d && p[2] == 0x3d),
+        "Unexpected instruction - this can happen when this is not compiled in .so (e.g. for single file)");
+
+    // At this point, `p` contains the instruction pointer and is pointing to the above opcodes.
+    // These opcodes are patched by the dynamic linker.
+    // Move beyond the opcodes that we have already checked above.
+    p += 3;
+
+    // The descriptor address is located at *p at this point. Read that and add
+    // it to the instruction pointer to locate the address of `ti` that will be used
+    // to pass to __tls_get_addr during execution.
+    // (p + 4) below skips the descriptor address bytes embedded in the instruction and
+    // add it to the `instruction pointer` to find out the address.
+    return *(uint32_t*)p + (p + 4);
+}
+
+// Generates sequence for accessing offset in TLS for osx/x64
+void* getThreadStaticsBaseOffset()
+{
+    uint8_t* p;
+    __asm__ (
+        "leaq  0(%%rip), %%rdx\n"
+        "movq  _t_ThreadStatics@TLVP(%%rip), %%rdi\n"
+        : "=d"(p)
+        );
+
+    return getThreadStaticDescriptor(p);
+}
+
+#else
+void* getThreadStaticDescriptor(uint8_t* p)
+{
+    _ASSERTE_MSG((p[0] == 0x66 && p[1] == 0x48 && p[2] == 0x8d && p[3] == 0x3d),
+        "Unexpected instruction - this can happen when this is not compiled in .so (e.g. for single file)");
+
+    // At this point, `p` contains the instruction pointer and is pointing to the above opcodes.
+    // These opcodes are patched by the dynamic linker.
+    // Move beyond the opcodes that we have already checked above.
+    p += 4;
+
+    // The descriptor address is located at *p at this point. Read that and add
+    // it to the instruction pointer to locate the address of `ti` that will be used
+    // to pass to __tls_get_addr during execution.
+    // (p + 4) below skips the descriptor address bytes embedded in the instruction and
+    // add it to the `instruction pointer` to find out the address.
+    return *(uint32_t*)p + (p + 4);
+}
+
+// Generates sequence for accessing offset in TLS for linux/x64
+// Note: This method is marked as NOINLINE, so we preserve `rbx
```

Comment:
This is not the correct way to solve this problem. The correct way to solve this problem is to specify rbx as register trashed by the inline assembly: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html . It is the `Clobbers` argument after the last `:`.

Reply:
> The correct way to solve this problem is to specify rbx as register trashed by the inline assembly

Yep, that's exactly I thought and added both `rbx` and `rdi` in the clobbers list. But with the presence of `rbx`, I got the error `asm-specifier for input or output variable conflicts with asm clobber list` for `rbx`. So I removed `rbx` and kept just `rdi`.

It was essentially overwriting `isGCType` and `threadStaticBaseOffset`. However, even with this change I am seeing similar failure although locally I found the build worked fine. I am trying to see locally again.